packages/lisk-api-client/test/resources/signatures.ts

```diff
@@ -12,36 +12,32 @@
  * Removal or modification of this copyright notice is prohibited.
  *
  */
-
-import APIResource from '../../src/api_resource';
-import SignaturesResource from '../../src/resources/signatures';
+import { expect } from 'chai';
+import { APIClient } from '../../src/api_client';
+import { APIResource } from '../../src/api_resource';
+import { SignaturesResource } from '../../src/resources/signatures';
+import { FakeApiClient } from '../types/types';
 
 describe('SignaturesResource', () => {
 	const defaultBasePath = 'http://localhost:1234';
 	const path = '/signatures';
 
-	let apiClient;
-	let resource;
+	let apiClient: FakeApiClient;
```

Comment:
If we could write an interface for `apiClient` class and implement it then there will be no need to create a fake type for testing and also it will be reusable across the package.

Reply:
@ishantiw Main reason for fake type is because many properties on APIClient class are not optional, but unneeded for testing purposes. As for creating interface for APIClient class, classes can be used as interfaces in typescript, but it may be cleaner to create interface and use separately? Thoughts @shuse2?

Reply:
Decision to use Classes as interface unless in multiple implementation cases as per discussion @shuse2 @ishantiw